server/controllers/user.js

```diff
@@ -25,7 +25,19 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
 
 exports.getUserBoards = asyncHandler(async (req, res, next) => {
   //@ts-ignore
-  const boards = await Board.find({ user: req.user.id });
+  const boards = await Board.find({ user: req.user.id }).populate({
+    path: "columns",
+    populate: {
+      path: "cards",
+      model: "card",
+    },
+  });
+  /* If no no boards, create a single template board */
+  if (!boards) {
+    const newBoard = await Board.create().createTemplateBoard();
+    boards.push(newBoard);
+    await boards.populate();
+  }
```

Comment:
I don't understand this, if a user deleted all their boards intentionally when they reload it will give them a new board regardless?

Reply:
You know, I added this to handle the case where a user had no boards (it was causing the front end to crash), but I've since added some null checks into the code so it can actually handle having zero boards. I'll take this out now.

Reply:
Thank you!